### PR TITLE
imx6ull-flash fixes

### PR DESCRIPTION
**Pull request by agkaminski**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (DTR).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/imx6ull-flash/flashdrv.c [hunk 1]
```diff
@@ -447,16 +447,20 @@ static void flashdrv_setDevClock(int dev, int state)
...
 	return dma;
 }
 
 
 void flashdrv_dmadestroy(flashdrv_dma_t *dma)
 {
-	munmap(dma, _PAGE_SIZE);
+	if (dma != MAP_FAILED && dma != NULL) {
+		munmap(dma, _PAGE_SIZE);
+	}
```
Comment (badochov) [inline storage/imx6ull-flash/flashdrv.c:463]: When can dma be null? Also I beleive that check for dma == MAP_FAILED is redundant as this should be checked just after performing mmap.

File: storage/imx6ull-flash/flashdrv.c [hunk 2]
```diff
@@ -447,16 +447,20 @@ static void flashdrv_setDevClock(int dev, int state)
 flashdrv_dma_t *flashdrv_dmanew(void)
 {
 	flashdrv_dma_t *dma = mmap(NULL, _PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_NULL, 0);
-	dma->last = NULL;
-	dma->first = NULL;
+	if (dma != MAP_FAILED) {
+		dma->last = NULL;
+		dma->first = NULL;
+	}
```
Comment (kemonats) [inline storage/imx6ull-flash/flashdrv.c:453]: A little beyond that PR but the `setup_flash_info()` function does not check the value returned by the `flashdrv_dmanew()` function.

File: storage/imx6ull-flash/flashdrv.c (hunk 1 above)
Comment (ziemleszcz) [reply storage/imx6ull-flash/flashdrv.c:463]: Take a look at `setup_flash_info()` for MAP_FAILED case. NULL check is probably added as protection against careless deinitialization.

**Review by ziemleszcz — APPROVED** [review]
LGTM (except kemonats comment)

File: storage/imx6ull-flash/flashdrv.c (hunk 2 above)
Comment (agkaminski) [reply storage/imx6ull-flash/flashdrv.c:453]: Fixed (somewhat), thank you

**Review by ziemleszcz — APPROVED** [review]

**Pull request merged**